tests/puppeteer/pages/commonPage.js

```diff
@@ -17,6 +19,7 @@ module.exports = class CommonPage {
    * @return textContent
    */
   async getTextContent(selector) {
+    await this.page.waitForSelector(selector, {visible: true});
```

Comment:
No need to add this here, I think you should wait for your selector after the click in function `generatePDFByDate`

Reply:
There is no validation message all the time, the error message is displayed only when there is no invoice in the selected date.

Reply:
Still, it's weird to have waitForSelector here, it will be duplicated in some functions